test/src/tests-identities-attributes.ts

```diff
@@ -1223,6 +1237,135 @@ describe('identities and attributes', function() {
+        const secondCallBody = JSON.parse(secondCall[1].body as unknown as string) as Batch;
+        expect(secondCallBody.events[0].event_type).to.equal('session_start', 'Second Call');
+
+        const thirdCall = fetchMock.calls()[2];
+        expect(thirdCall[0].split('/')[6]).to.equal('events');
+        const thirdCallBody = JSON.parse(thirdCall[1].body as unknown as string) as Batch;
+        expect(thirdCallBody.events[0].event_type).to.equal('application_state_transition', 'Third Call');
+
+        const fourthCall = fetchMock.calls()[3];
+        expect(fourthCall[0].split('/')[6]).to.equal('events');
+        const fourthCallBody = JSON.parse(fourthCall[1].body as unknown as string) as Batch;
+        expect(fourthCallBody.events[0].event_type).to.equal('user_identity_change', 'Fourth Call');
+
+        const fifthCall = fetchMock.calls()[4];
+        expect(fifthCall[0].split('/')[6]).to.equal('events');
+        const fifthCallBody = JSON.parse(fifthCall[1].body as unknown as string) as Batch;
+        const fifthCallEvent = fifthCallBody.events[0] as CustomEvent;
+        expect(fifthCallEvent.event_type).to.equal('custom_event', 'Fifth Call');
+        expect(fifthCallEvent.data.event_name).to.equal('Test Event 1', 'Fifth Call');
+
+        const sixthCall = fetchMock.calls()[5];
+        expect(sixthCall[0].split('/')[6]).to.equal('events');
+        const sixthCallBody = JSON.parse(sixthCall[1].body as unknown as string) as Batch;
+        const sixthCallEvent = sixthCallBody.events[0] as CustomEvent;
+        expect(sixthCallBody.events[0].event_type).to.equal('custom_event', 'Sixth Call');
+        expect(sixthCallEvent.data.event_name).to.equal('Test Event 2', 'Sixth Call');
+
+        const seventhEvent = fetchMock.calls()[6];
+        expect(seventhEvent[0].split('/')[6]).to.equal('events');
+        const seventhEventBody = JSON.parse(seventhEvent[1].body as unknown as string) as Batch;
+        const seventhEventEvent = seventhEventBody.events[0] as CustomEvent;
+        expect(seventhEventBody.events[0].event_type).to.equal('custom_event', 'Seventh Call');
+        expect(seventhEventEvent.data.event_name).to.equal('Test Event 3', 'Seventh Call');
+    });
+
+    it('should order user identity change events before logging any events that are in the ready queue', async () => {
+
+        mParticle._resetForTests(MPConfig);
+        fetchMock.resetHistory();
+
+        // Clear out before each init call
+        await waitForCondition(hasBeforeEachCallbackReturned);
+
+        window.mParticle.config.identifyRequest = {
+            userIdentities: {
+                email: 'initial@gmail.com',
+            },
+        };
+
+        mParticle.init(apiKey, window.mParticle.config);
+
+        mParticle.logEvent('Test Event 1');
+        mParticle.logEvent('Test Event 2');
+        mParticle.logEvent('Test Event 3');
+
+        expect(mParticle.getInstance()._preInit.readyQueue.length).to.equal(3);
```

Comment:
this test is identical to the test above here but without the readyqueue assertion.  however, i imagine that the above test would also pass if you put this assertion there.  is there a way to differentiate the two?

Reply:
I think the two tests should represent the workflow with and without the ready queue, so technically they would be nearly identical. I made a change to the other test to make it a bit more clear. How does that look?